### DISCUSSION TITLE
How to stream request body?

### DISCUSSION THREAD

**Discussion by gaby** [earthboundkid/requests#85] [Support] [answered]
I have a server that accepts a streamed request body. How can I stream a local file to the server using this library? I see there's a function for streaming from Server to Local but not from Local to Server.

This can be done using the folliwing: 

```console
curl -X POST --data-binary @/path/to/large/file localhost:3000
```

In Python, this is done by passing the file handler to the `data` param in `requests`:

```python
with open(filepath, 'rb') as f:
    r = requests.post(url, data=f)
```

**Comment by earthboundkid** [comment, marked as answer]
I’m on mobile, but you can use https://pkg.go.dev/github.com/carlmjohnson/requests#Builder.BodyReader or https://pkg.go.dev/github.com/carlmjohnson/requests#Builder.BodyWriter depending on the source.

**Reply by gaby** [reply]
Thank you, that's exactly what I was looking for :-)

**Reply by earthboundkid** [reply]
For files specifically, BodyFile streams the file.
